### PR TITLE
Add blocklist functionality for applicants

### PR DESCRIPTION
**Pull request by prpcodes**
- Add BlockUserButton to applications
- Add "Blocked" to the user application status
- Add Blocked to the Status enum
- Upon opening the applyForm, check if the last application was blocked
- Display a toast that prompts the user to contact us
- Send them to the contact form

### CODE REVIEW

File: src/components/layout/playground/applyForm.tsx [hunk 1]
```diff
@@ -319,6 +319,16 @@ const MainForm: React.FC<RequestProps> = ({ request }) => {
   useOnce(
     () => {
       if (!lastApplication) return;
+
+      if (lastApplication?.status === 'Blocked') {
```
Comment (MatsThieme) [inline src/components/layout/playground/applyForm.tsx]: `import { Status } from '@prisma/client';`
Please import the enum and use Status.Blocked

File: src/pages/playground/admin/applications.tsx [hunk 2]
```diff
@@ -150,6 +151,21 @@ const AdminPage: NextPage = ({}) => {
...
+                                confirm(
+                                  `Are you sure you want to block ${app.name} from taking on future applications?`
+                                )
+                              ) {
+                                mutate({ id: app.id, status: 'Blocked' });
+                              }
+                            }}
+                          >
+                            ⛔ Block
+                          </DenyButton>
```
Comment (MatsThieme) [inline src/pages/playground/admin/applications.tsx]: grafik
that's a bit cut off, i would use two lines.

File: src/pages/playground/admin/applications.tsx [hunk 3]
```diff
@@ -150,6 +151,21 @@ const AdminPage: NextPage = ({}) => {
...
+                          <DenyButton
+                            className="w-full"
+                            disabled={isLoading}
+                            onClick={() => {
+                              if (
+                                confirm(
+                                  `Are you sure you want to block ${app.name} from taking on future applications?`
+                                )
+                              ) {
+                                mutate({ id: app.id, status: 'Blocked' });
```
Comment (MatsThieme) [inline src/pages/playground/admin/applications.tsx]: `import { Status } from '@prisma/client';`
Please import the enum and use Status.Blocked

File: src/components/layout/playground/applyForm.tsx [hunk 4]
```diff
@@ -319,6 +319,16 @@ const MainForm: React.FC<RequestProps> = ({ request }) => {
...
+        return toast.error(
+          'Please contact us before applying for more requests.',
+          {
+            onClose: () =>
+              router.push('/contact', undefined, { shallow: true }),
+          }
```
Comment (MatsThieme) [inline src/components/layout/playground/applyForm.tsx]: A toast can easily be missed, to allow the user to navigate on their own, please show this message where the form is currently placed and put a contact button below it. Like this 
grafik

example code
```jsx
  if (isBlocked) {
    return (
      <>
        <h1 className="text-3xl">
          ⚠️ Please contact us before applying for more requests.
        </h1>
        <DarkButton className="w-fit m-auto mt-5" href="/contact">
          Contact
        </DarkButton>
      </>
    );
  }
```

**Review by MatsThieme — CHANGES_REQUESTED** [review]

File: src/components/layout/playground/applyForm.tsx [hunk 5]
```diff
@@ -1,659 +1,659 @@
-import { DateTime } from 'luxon';
```
Comment (MatsThieme) [inline src/components/layout/playground/applyForm.tsx]: Please fix the line endings in this file

File: src/components/layout/playground/applyForm.tsx [hunk 6]
```diff
@@ -621,15 +621,43 @@ const MainForm: React.FC<RequestProps> = ({ request }) => {
   );
 };
 
+const ApplicantBlocked: React.FC = () => {
```
Comment (MatsThieme) [inline src/components/layout/playground/applyForm.tsx]: Please rename to ```RequestApplicationBlocked```

File: src/components/layout/playground/applyForm.tsx [hunk 7]
```diff
@@ -621,15 +621,43 @@ const MainForm: React.FC<RequestProps> = ({ request }) => {
...
+  return (
+    <>
+      <div className="text-5xl">⚠️</div>
+      <h1 className="text-2xl my-5">
+        Please contact us before applying for more requests.
```
Comment (MatsThieme) [inline src/components/layout/playground/applyForm.tsx]: Please change to ```Please contact us before submitting another application.```

**Review by MatsThieme — CHANGES_REQUESTED** [review]

**Review by MatsThieme — APPROVED** [review]

**Comment by MatsThieme** [comment]
Nice work 😃
Please squash your commits when merging to main.

**Comment by prpcodes** [comment]
> Nice work 😃 Please squash your commits when merging to main.

Thanks, will do 😃

**Pull request merged**